2020-epc-dev/demo-embed-arcgis/src/components/ContextPane.js

```diff
@@ -52,20 +72,20 @@ function ContextPane(){
             Routing service<br/>
             ArcGIS API for JavaScript<br/>
           </ContextP>
-          <Button clearWhite onClick={_ => history.push('/lbs')}>LBS app</Button>
+          <Button clearWhite onClick={_ => {ClearLogin(); history.push('/lbs')}}>LBS app</Button>
         </CardContent>
       </ContextCard>
       <ContextCard bar={infrActive ? 'white' : null}>
         <CardContent>
-        <ContextTitle active={infrActive}>Infrastructure</ContextTitle>
+        <ContextTitle active={infrActive}>Identity</ContextTitle>
           <ContextP active={infrActive}>
             Hosted feature services & views<br/>
-            Basic User Types<br/>
+            User Types<br/>
           </ContextP>
           <ButtonGroup>
-            <Button clearWhite={path !== '/user/request'} white={path === '/user/request'} fullWidth onClick={_ => history.push('/user/request')}>Form</Button>
-            <Button clearWhite={path !== '/user/dispatch'} white={path === '/user/dispatch'} fullWidth onClick={_ => history.push('/user/dispatch')}>Dispatch</Button>
-            <Button clearWhite={path !== '/user/worker'} white={path === '/user/worker'} fullWidth onClick={_ => history.push('/user/worker')}>Worker</Button>
+            <Button clearWhite={path !== '/user/request'} white={path === '/user/request'} fullWidth onClick={_ => {ClearLogin(); history.push('/user/request')}}>Form</Button>
+            <Button clearWhite={path !== '/user/dispatch'} white={path === '/user/dispatch'} fullWidth onClick={_ => {DispatchLogin(); history.push('/user/dispatch')}}>Dispatch</Button>
+            <Button clearWhite={path !== '/user/worker'} white={path === '/user/worker'} fullWidth onClick={_ => {WorkerLogin(); history.push('/user/worker')}}>Worker</Button>
```

Comment:
Remove these functions.

As an FYI, the React paradigm is to have values react to the "state" of the app, rather than setting all values explicitly. Here you know based on the route path what the user is looking at (even if it's not explicitly React's `state` object), so you can define the username / password based on that without calling setting functions on click:
```js
function ContextPane(){
  // ...
  const app = path[0];
  let username, password;
  if (app === 'worker') {
    username = 'worker_username';
    password = 'worker_password';
  } else if (app === 'dispatch'){
    username = 'dispatch_username';
    password = 'dispatch_password';
  }
  // ...
}
```